File: adapter.go
package main

import "adapter"

func main() {
	zhangsan := adapter.Person{"hello"}
	zhangsan.Express() // hello

	lisi := adapter.Mute{"I want to express!"}
	notebook := adapter.NoteBook{lisi}
	notebook.Express() // I want to express!
}

File: java/src/Adapter.java
import pattern.adapter.DeafMan;
import pattern.adapter.NormalPerson;
import pattern.adapter.SpeakAdapter;

public class Adapter {
    public static void main(String[] args) {
        NormalPerson zhangsan = new NormalPerson("zhangsan");
        zhangsan.speak("hi"); // zhangsan speak:hi

        DeafMan lisi = new DeafMan("lisi");
        SpeakAdapter adapter = new SpeakAdapter();
        adapter.setMan(lisi);
        adapter.speak("hello"); // lisi speak:hello
    }
}
